src/platforms/hosted/serial_win.c

```diff
@@ -266,16 +364,66 @@ bool platform_buffer_write(const void *const data, const size_t length)
 	DEBUG_WIRE("%s\n", buffer);
 	DWORD written = 0;
 	for (size_t offset = 0; offset < length; offset += written) {
-		if (!WriteFile(port_handle, buffer + offset, length - offset, &written, NULL)) {
-			DEBUG_ERROR("Serial write failed %lu, written %zu\n", GetLastError(), offset);
-			return false;
+		if (port_handle != INVALID_HANDLE_VALUE) {
+			if (!WriteFile(port_handle, buffer + offset, length - offset, &written, NULL)) {
+				DEBUG_ERROR("Serial write failed %lu, written %zu\n", GetLastError(), offset);
+				return false;
+			}
+		} else if (network_socket != INVALID_SOCKET) {
+			int network_written = send(network_socket, buffer + offset, length - offset, 0);
+			if (network_written == SOCKET_ERROR) {
+				DEBUG_ERROR("Network write failed %d, written %zu\n", WSAGetLastError(), offset);
+				return false;
+			}
+			written = network_written;
 		}
 	}
 	return true;
 }
 
+static ssize_t bmda_read_more_socket_data()
```

Comment:
This needs its parens set to `(void)`. While this compiles it [accidentally invokes the non-prototype function declaration](https://en.cppreference.com/w/c/language/function_declaration#Syntax) syntax which has undesirable semantics in this context.